docs/rules/on-function-explicit-return-type.md

```diff
@@ -1 +1,43 @@
 # on-function-explicit-return-type
+
+> This rule disallows using the `on` reducer function without providing an explicit return type.
+
+## Rule Details
+
+When we use the `on` function to create reducers, we usually copy the state into a new object, and then add the properties that are being modified after that certain action. This may result in unexpected typing problems: we can add new properties into the state that did not exist previously, and TypeScript would not infer the problem. The solution is to provide an explicit return type to the `on` function callback.
```

Comment:
```suggestion
When we use the `on` function to create reducers, we usually copy the state into a new object, and then add the properties that are being modified after that certain action. This may result in unexpected typing problems, we can add new properties into the state that did not exist previously. TypeScript doesn't see this as a problem, and might change the state's interface. The solution is to provide an explicit return type to the `on` function callback.
```